### PR TITLE
HUM-312 : Nursery / object stabilization for replicated objects

### PR DESCRIPTION
**Pull request by dpgoetz**

### CODE REVIEW

**Comment by dpgoetz** [comment]
WIP!! do not merge

File: objectserver/nurseryobjeng.go [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+	}
+	if o.file != nil {
+		defer o.file.Close()
+		o.file = nil
+	}
+	return nil
+}
+
+func (o *nurseryObject) Stabilize() error {
+	if fs.Exists(o.stableHashDir) {
```
Comment (corystone) [inline objectserver/nurseryobjeng.go]: Nitpick:
if !exists { create() }
lock()

File: objectserver/nurserystabilizer.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+		if device.Ip == nrd.dev.Ip && device.Port == nrd.dev.Port && device.Device == device.Device {
+			continue
+		}
+		url := fmt.Sprintf("http://%s:%d/%s/%d%s", device.Ip, device.Port, device.Device, partition, common.Urlencode(metadata["name"]))
+		req, err := http.NewRequest("HEAD", url, nil)
+		req.Header.Set("X-Backend-Storage-Policy-Index", strconv.FormatInt(int64(nrd.policy), 10))
+		req.Header.Set("User-Agent", "nursery-stabilizer")
+		resp, err := nrd.client.Do(req)
+
+		if err == nil && (resp.StatusCode/100 == 2 || resp.StatusCode == 404) &&
```
Comment (corystone) [inline objectserver/nurserystabilizer.go]: I'm not sure I understand why 404 is ok here.

Comment (dpgoetz) [reply objectserver/nurserystabilizer.go]: tombstones will 404

Comment (corystone) [reply objectserver/nurserystabilizer.go]: What about "I don't have that object at all"? Isn't that important here?

Comment (dpgoetz) [reply objectserver/nurserystabilizer.go]: if you don't have it at all then goodNodes won't be incremented and the object will be skipped. it relies on replication to populate the other nodes.

File: objectserver/nurseryobjeng.go (hunk 1 above)
Comment (dpgoetz) [reply objectserver/nurseryobjeng.go]: done

**Pull request merged**